dvc/utils/strictyaml.py

```diff
@@ -6,24 +6,43 @@
 
 Not to be confused with strictyaml, a python library with similar motivations.
 """
-from typing import TYPE_CHECKING, Any, Callable, List, TypeVar
+import typing
+from contextlib import suppress
+from typing import TYPE_CHECKING, Any, Callable, List, Optional, Tuple, TypeVar
 
-from dvc.exceptions import DvcException, PrettyDvcException
+import regex as re
+
+from dvc.exceptions import PrettyDvcException
 from dvc.utils.serialize import (
     EncodingError,
     YAMLFileCorruptedError,
     parse_yaml,
+    parse_yaml_for_update,
 )
 
 if TYPE_CHECKING:
     from rich.syntax import Syntax
     from rich.text import Text
     from ruamel.yaml import StreamMark
+    from voluptuous import MultipleInvalid
 
     from dvc.fs.base import BaseFileSystem
 
 
 _T = TypeVar("_T")
+merge_conflict_marker = re.compile("^([<=>]{7}) .*$", re.MULTILINE)
+
+
+def make_relpath(path: str) -> str:
+    import os
+
+    from dvc.utils import relpath
+
+    rel = relpath(path)
+    prefix = ""
+    if not rel.startswith(".."):
+        prefix = "./" if os.name == "posix" else ".\\"
```

Comment:
```
'dvc.yaml' is corrupted.
```

^ This kind of message was confusing, at least I find it to be. As it may be talking about `dvc.yaml` file in a different place in the workspace or in the user's working directory. Added `./dvc.yaml`/`.\dvc.yaml` to make it obvious that it's relative path.

Reply:
Absolute path may be quite lengthy. We don't have guidelines for that, but for UI, we have always used relpaths.

Reply:
That works, should we document that as a guideline?